lib/active_admin/application.rb

```diff
@@ -150,8 +150,8 @@ def routes(rails_router)
     AbstractController::Callbacks::ClassMethods.public_instance_methods.
       select { |m| m.match(/_action\z/) }.each do |name|
       define_method name do |*args, &block|
-        controllers_for_filters.each do |controller|
-          controller.public_send name, *args, &block
+        ActiveSupport.on_load(:active_admin_controller) do
+          public_send name, *args, &block
```

Comment:
Just curious since this previously was looping for each controller, why is that no longer necessary?

Reply:
Thanks! I think it makes sense now. From there though I didn't understand why the separate `run_load_hooks` calls using `..._base` or `..._devise` when they have the same base object argument?

Reply:
Yes, I have the same question as @javierjulio. Only running hooks we registered ourselves (the ones tagged as `:active_admin_controller`) seems to fix the deprecation for me. Why do we need to run the others?

Reply:
Sorry for the confusion. `..._base` and `..._devise` are not required to fix the issue. 
I was inspired by Rails and they provide hooks for controller, API controllers and base controller. I thought it was a good idea to leave a door open to define hooks for only base controller or devise controllers exclusively (without an if inside `:active_admin_controller`.
I just change my mind and remove them. They are a source of confusion and aren't needed right now.

Reply:
No worries at all. Thank you so much for clarifying @mgrunberg!